src/apollo/config.js

```diff
@@ -2,27 +2,70 @@ import fetch from 'cross-fetch';
 import {
   ApolloClient,
   InMemoryCache,
-  createHttpLink,
-  split,
+  HttpLink,
+  ApolloLink,
+  Observable,
 } from '@apollo/client';
-import { getMainDefinition } from '@apollo/client/utilities';
+import { onError } from '@apollo/client/link/error';
 import { sanitizedPathname } from '../utils';
+import loadJsonData from '../store/load-data';
 
-const httpLink = createHttpLink({
-  // our graphql endpoint, normally here: http://localhost:4141/graphql
-  uri: `${sanitizedPathname()}graphql`,
+const apiBaseUrl = `${sanitizedPathname()}graphql`;
+
+// HTTP link for GraphQL API calls
+const httpLink = new HttpLink({
+  uri: apiBaseUrl,
   fetch,
 });
 
-const splitLink = split(({ query }) => {
-  const definition = getMainDefinition(query);
+// Error handling link
+const errorLink = onError(
+  ({ graphQLErrors, networkError, operation, forward, response }) => {
+    if (networkError) {
+      console.error(`[Network error]: ${networkError}`);
+
+      // Try reading from static file if network error occurs
```

Comment:
Instead of checking for network error and then reading from a static file path, can we check for the host url ? Like if the url host is the demo site, then fetch from file instead of making a network call ? This seems like a round-trip which is not needed if we already know the request is made from demo site

Reply:
But we still cant ignore below errors, right ?

<img width="1180" alt="Screenshot 2024-08-09 at 3 35 57 p m" src="https://github.com/user-attachments/assets/386a5560-54a8-4f4a-80d2-6e11acb01ea3">

Reply:
I am thinking doing this but first I want to test it out with demo.kedro.org. So I will make this change in a follow up PR where I update the github workflow to 'public-kedro-viz' actions